pkg/network/sriov_test.go

```diff
@@ -0,0 +1,156 @@
+package network
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	osv1 "github.com/openshift/api/operator/v1"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+var _ = Describe("Testing sriov", func() {
+	Describe("validateSriov", func() {
+		_validateSriov := func(
+			sriovRequested bool,
+			openshiftNetworkOperatorRunning bool,
+			openshiftSriovEnabled bool,
+		) []error {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{}
+			if sriovRequested {
+				conf.Sriov = &opv1alpha1.Sriov{}
+			}
+
+			var openshiftNetworkConfig *osv1.Network
+			if openshiftNetworkOperatorRunning {
+				openshiftNetworkConfig = &osv1.Network{}
+				if openshiftSriovEnabled {
+					openshiftNetworkConfig.Spec.AdditionalNetworks = append(openshiftNetworkConfig.Spec.AdditionalNetworks, osv1.AdditionalNetworkDefinition{
+						Type:         osv1.NetworkTypeRaw,
+						Name:         "sriov-network-name",
+						RawCNIConfig: "{\"type\": \"sriov\"}",
+					})
+				}
+			}
+
+			return validateSriov(conf, openshiftNetworkConfig)
+		}
+
+		Context("when configuration is empty", func() {
+			sriovRequested := false
+			openshiftNetworkOperatorRunning := false
+			openshiftSriovEnabled := false
+			It("should pass", func() {
+				errorList := _validateSriov(sriovRequested, openshiftNetworkOperatorRunning, openshiftSriovEnabled)
+				Expect(errorList).To(BeEmpty())
+			})
+		})
+
+		Context("when openshfit network operator is running", func() {
+			openshiftNetworkOperatorRunning := true
+			Context("and has sriov enabled", func() {
+				openshiftSriovEnabled := true
+				Context("and configuration is empty", func() {
+					sriovRequested := false
+					It("should pass", func() {
+						errorList := _validateSriov(sriovRequested, openshiftNetworkOperatorRunning, openshiftSriovEnabled)
+						Expect(errorList).To(BeEmpty())
+					})
+				})
+				Context("and configuration requests sriov", func() {
+					sriovRequested := true
+					It("should fail with an error message", func() {
+						errorList := _validateSriov(sriovRequested, openshiftNetworkOperatorRunning, openshiftSriovEnabled)
+						Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %v", errorList)
+						Expect(errorList[0].Error()).To(Equal("SR-IOV has been requested, but it's not compatible with OpenShift Cluster Network Operator SR-IOV support"))
```

Comment:
(no action required) personally not a fan of matching against exact error messages but I guess it's subjective

Reply:
Is there something like "error type" in Go? Like checking Exception class in Python?

Reply:
I don't think it can be directly mimicked. I don't know if there is a good solution. Is the exact message important enough to check in the first place?

Reply:
I'd say it does no harm. I'd rather change the message at two places than have ambiguous test.